lnwallet/channel.go

```diff
@@ -8445,16 +8445,27 @@ func (lc *LightningChannel) MaxFeeRate(maxAllocation float64) chainfee.SatPerKWe
 	availableBalance, weight := lc.availableBalance()
 
 	oldFee := lc.localCommitChain.tip().feePerKw.FeeForWeight(weight)
-
 	// baseBalance is the maximum amount available for us to spend on fees.
 	baseBalance := availableBalance.ToSatoshis() + oldFee
 
+	// In case our local channel balance is drained, we make sure we do not
+	// decrease the fee rate below the old fee rate. This could lead to a
+	// scenario where we lower the commitment fee rate as low as the fee
+	// floor although current fee rates are way higher. The maximum fee
+	// we allow should not be smaller then the current fee, the decrease
+	// in fees should when the mempool reports lower fee level rather than
+	// us decreasing in local balance.
 	maxFee := float64(baseBalance) * maxAllocation
+	maxFee = math.Max(maxFee, float64(oldFee))
 
 	// Ensure the fee rate doesn't dip below the fee floor.
-	maxFeeRate := maxFee / (float64(weight) / 1000)
-	return chainfee.SatPerKWeight(
-		math.Max(maxFeeRate, float64(chainfee.FeePerKwFloor)),
+	maxFeeRate := chainfee.SatPerKWeight(maxFee / (float64(weight) / 1000))
+
+	// Normally the old fee rate is always greater than the fee floor
+	// otherwise it would not have been selected but we keep this check in
```

Comment:
I don't think we should do this check anymore as we are using at least `oldFeeRate` and, if that value is below fee floor, then it means there's a bug in other subsystems, such as fee estimator - in that case we should fix it there. Defaulting to fee floor here will hide that bug. Also means we can remove `MaxSatPerKweight` for now?

Reply:
renamed to `currentFee` instead of `oldFee` and also removed the check.